### PR TITLE
Project many-to-many hibernate implementation.

### PR DESCRIPTION
**Pull request by Karthick9917**

### CODE REVIEW

File: com/ideas2it/employeeManagementSystem/controller/EmployeeController.java [hunk 1]
```diff
@@ -96,11 +96,22 @@ public EmployeeDTO getEmployeeById (int employeeId) {
...
     }
 
+    /**
+     * Passing the integer and its return the object
+     * based on the given id.
+     * @param projectId - passing the given integer.
+     * @return return the object based on the id.
+     */
+    public ProjectDTO getProjectById(int projectId) {
+        return employeeService.getProjectById(projectId);
```
Comment (ponnuvelgandhi) [inline com/ideas2it/employeeManagementSystem/controller/EmployeeController.java]: Use Project Service to get the project by project Id

File: com/ideas2it/employeeManagementSystem/controller/ProjectController.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+     * @param date - for check the date is valid or not
+     * @return true for false once the input is valid.
+     */
+    public boolean getDate(String date) {
+        return projectService.validateDate(date);
+    }
+
+
+    /**
+     * Passing the string and its return the true or false
```
Comment (ponnuvelgandhi) [inline com/ideas2it/employeeManagementSystem/controller/ProjectController.java:50]: Write proper comments

File: com/ideas2it/employeeManagementSystem/controller/ProjectController.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    /**
+     * Passing the integer and its return the object
+     * based on the given id.
+     * @param employeeId - passing the given integer.
+     * @return return the object based on the id.
+     */
+    public EmployeeDTO getEmployeeById(int employeeId) {
+        return projectService.getEmployeeById(employeeId);
```
Comment (ponnuvelgandhi) [inline com/ideas2it/employeeManagementSystem/controller/ProjectController.java]: Use employee service here

File: com/ideas2it/employeeManagementSystem/dao/impl/EmployeeDao.java [hunk 4]
```diff
@@ -47,9 +47,11 @@ public boolean createEmployeeDetails (Employee employee) throws EmsException {
             session.getTransaction().commit();
         } catch (HibernateException hibernateException) {
             logger.error(hibernateException.getMessage());
-            throw new EmsException(EmployeeConstants.NOT_ADDED_MESSAGE);
+            throw new EmsException(Constants.NOT_ADDED_MESSAGE);
```
Comment (ponnuvelgandhi) [inline com/ideas2it/employeeManagementSystem/dao/impl/EmployeeDao.java]: Use proper constants

File: com/ideas2it/employeeManagementSystem/dao/impl/ProjectDAO.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+     */
+    public List<Project> getProjectsByName(String projectName) throws EmsException {
+        List<Project> searchedProjectList = new ArrayList<Project>();
+        Session session = null;
+        try {
+            session = connectionUtil.getConnection();
+            session.beginTransaction();
+            Criteria criteria = session.createCriteria(Project.class);
+            searchedProjectList = (List<Project>) criteria.add(Restrictions.like("projectName", (projectName + "%"))).list();
+            session.getTransaction().commit();
```
Comment (ponnuvelgandhi) [inline com/ideas2it/employeeManagementSystem/dao/impl/ProjectDAO.java]: While getting the record, no need to commit the transaction

File: logs/app.log [hunk 6]
```diff
@@ -0,0 +1,12 @@
+2022-10-28 16:57:40 INFO o.h.Version:46 HHH000412: Hibernate Core {5.3.1.Final}
```
Comment (ponnuvelgandhi) [inline logs/app.log]: Should not commit log file

**Review by ponnuvelgandhi — CHANGES_REQUESTED** [review]

File: com/ideas2it/employeeManagementSystem/controller/EmployeeController.java (hunk 1 above)
Comment (Karthick9917) [reply com/ideas2it/employeeManagementSystem/controller/EmployeeController.java]: Correction resolved

File: com/ideas2it/employeeManagementSystem/controller/ProjectController.java (hunk 3 above)
Comment (Karthick9917) [reply com/ideas2it/employeeManagementSystem/controller/ProjectController.java]: Correction resolved

File: com/ideas2it/employeeManagementSystem/controller/ProjectController.java (hunk 2 above)
Comment (Karthick9917) [reply com/ideas2it/employeeManagementSystem/controller/ProjectController.java:50]: Correction resolved

File: com/ideas2it/employeeManagementSystem/dao/impl/ProjectDAO.java (hunk 5 above)
Comment (Karthick9917) [reply com/ideas2it/employeeManagementSystem/dao/impl/ProjectDAO.java]: Correction resolved

File: com/ideas2it/employeeManagementSystem/dao/impl/EmployeeDao.java (hunk 4 above)
Comment (Karthick9917) [reply com/ideas2it/employeeManagementSystem/dao/impl/EmployeeDao.java]: Correction resolved

File: logs/app.log (hunk 6 above)
Comment (Karthick9917) [reply logs/app.log]: Correction resolved

**Review by aruni2isankar — APPROVED** [review]

**Pull request merged**